### ISSUE TITLE
NoInject and noninjectable have different behaviour with default arguments

### ISSUE THREAD

**Issue by erikced** [alecthomas/injector#154]
When parameters are type-hinted with `NoInject` injector raises if there is more than one `NoInject` argument and at least one of them has a default argument and no explicit value is provided (i.e. one would expect that the default is used), this behaviour differs from `noninjectable` which handles it just fine.

With Injector 0.18.3 / Python 3.7.2
```python
from injector import Injector, NoInject, inject, noninjectable

@inject
def test_1(value_1: NoInject[int] = 1):
    return value_1

@inject
def test_2(value_1: int, value_2: NoInject[int] = 2):
    return value_1, value_2

@inject
def test_3(value_1: NoInject[int], value_2: NoInject[int] = 2):
    return value_1, value_2

@inject
@noninjectable('value_1', 'value_2')
def test_4(value_1: int, value_2: int = 2):
    return value_1, value_2

i = Injector()
i.call_with_injection(test_1)  # -> 1
i.call_with_injection(test_2, kwargs={'value_1': 1})  # -> (1, 2)
i.call_with_injection(test_3, kwargs={'value_1': 1}) # -> Error: UnknownProvider: couldn't determine provider for typing_extensions.Annotated[int, <object object at ... >]
i.call_with_injection(test_4, kwargs={'value_1': 1}) # -> (1, 2)

**Comment by jstasiak** [comment]
Thanks for the detailed report! Just so it's clear, the issue is with `test_3` only? The behavior in other tests is as you expect?

**Comment by erikced** [comment]
Yes, exactly.

**Comment by jstasiak** [comment]
#155 resolves this and version 0.18.4 that includes it is out.

**Comment by erikced** [comment]
> #155 resolves this and version 0.18.4 that includes it is out.

That was fast! Thank you.

**Comment by jstasiak** [comment]
No worries, enjoy. :)
